src/Reducer/userReducer.js

```diff
@@ -0,0 +1,57 @@
+export const Reducer = (state, action) => {
+  switch (action.type) {
+    case "Wishlist":
+      if (state.wishlist.length === 0) {
+        return { ...state, wishlist: [...state.wishlist, action.payload] };
+      } else {
+        return {
+          ...state,
+          wishlist: state.wishlist
+            .filter((item) => item.id !== action.id)
+            .concat(action.payload),
+        };
+      }
+    case "Remove_From_Wishlist":
+      return {
+        ...state,
+        wishlist: state.wishlist.filter((item) =>
+          item.id === action.payload ? false : true
```

Comment:
`filter(item=>item.id===action.payload)` will work fine,  no need to return `true` or `false` explicitly